magicblock-account-cloner/src/remote_account_cloner_worker.rs

```diff
@@ -442,12 +443,21 @@ where
         stage: ValidatorStage,
     ) -> AccountClonerResult<AccountClonerOutput> {
         let updated_clone_output = self.do_clone(pubkey, stage).await?;
-        self.last_clone_output
-            .write()
-            .expect("RwLock of RemoteAccountClonerWorker.last_clone_output is poisoned")
-            .insert(*pubkey, updated_clone_output.clone());
-        if let Ok(map) = self.last_clone_output.read() {
-            metrics::set_cached_clone_outputs_count(map.len());
+        // Do not cache non-existent accounts
+        let should_cache = match &updated_clone_output {
+            AccountClonerOutput::Unclonable { reason, .. } => {
+                !matches!(reason, AccountClonerUnclonableReason::DoesNotExist)
```

Comment:
`!matches!` 

What kind of Rust sorcery is that? Seeing such code for the first time.

Reply:
ah, nevermind! I see it now. The other branch helped me to decipher it. 

It's a boolean expression and the first `!` is a negation operation! Wow!